indico/modules/events/registration/fields/sessions.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from marshmallow import ValidationError, fields, validate
+
+from indico.core.db.sqlalchemy import db
+from indico.modules.events.registration.fields.base import RegistrationFormFieldBase
+from indico.modules.events.registration.models.registrations import RegistrationData
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.util.i18n import _
+
+
+class SessionsField(RegistrationFormFieldBase):
+    name = 'sessions'
+    mm_field_class = fields.List
+    mm_field_args = (fields.Integer,)
+    setup_schema_fields = {
+        'display': fields.String(load_default='expanded'),
+        'minimum': fields.Integer(load_default=0, validate=validate.Range(0)),
+        'maximum': fields.Integer(load_default=0, validate=validate.Range(0)),
+    }
```

Comment:
You need a real schema for this, see `NumberFieldDataSchema` as an example.
Otherwise you cannot validate that max>=min...